azure/services/virtualmachines/virtualmachines.go

```diff
@@ -76,11 +77,16 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	defer cancel()
 
 	vmSpec := s.Scope.VMSpec()
+	if vmSpec == nil {
+		return nil
+	}
 
 	result, err := s.CreateResource(ctx, vmSpec, serviceName)
 	s.Scope.UpdatePutStatus(infrav1.VMRunningCondition, serviceName, err)
 	// Set the DiskReady condition here since the disk gets created with the VM.
-	s.Scope.UpdatePutStatus(infrav1.DisksReadyCondition, serviceName, err)
+	if len(s.Scope.DiskSpecs()) > 0 {
```

Comment:
I don't think we even need to check this here. Since the disk is getting created as part of the VM, it's ready if the VM is ready.